source/deprecations/v2.x.html.md

```diff
@@ -753,3 +753,92 @@ component.appendTo('.my-component-wrapper');
 ```
 
 Note that both APIs are private, so no public API is been deprecated here.
+
+
+#### `{{render` helper
+
+##### until: 3.0.0
+##### id: ember-template-compiler.deprecate-render
+
+Using the `{{render` helper is deprecated in favor of using components.
+Please refactor to a component and invoke thusly:
+
+For example, if you had:
+
+```hbs
+{{render 'my-sidebar'}}
+```
+
+```hbs
+{{! app/templates/my-sidebar.hbs }}
+<p>template stuff here</p>
+```
+
+```js
+// app/controllers/my-sidebar.js
+export default Ember.Controller.extend({
+});
+```
+
+Would be refactored to:
+
+```hbs
+{{my-sidebar}}
+```
+
+```hbs
+{{! app/templates/components/my-sidebar.hbs }}
+<p>template stuff here</p>
+```
+
+```js
+// app/components/my-sidebar.js
+export default Ember.Component.extend({
+});
+```
+
+#### Rendering into a {{render}} helper that resolves to an {{outlet}}.
+
+##### until: 3.0.0
+##### id: ember-routing.top-level-render-helper
+
+For example, if you had:
```

Comment:
@josemarluedke I don't think anyone is very familiar with this API :-) Hence the strooong motivation to remove it :-)

I took a pass at an updated section [in this gist](https://gist.github.com/mixonic/b7a1051eaffa8db62027c4cfdbe1d2f0). Please give it a review, I may have missed something. It should be a good start to replace this section though, if you can merge that draft into this after review that seems good 👍